decidim-proposals/lib/tasks/proposals/upgrade/decdim_proposal_upgrade_tasks.rake

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :proposals do
+    namespace :upgrade do
+      desc "Removes all proposal valuator records of which the role assignment does not exists"
+      task remove_valuator_orphan_records: :environment do
+        Decidim::Proposals::ValuationAssignment
+          .where(valuator_role_type: "Decidim::ParticipatoryProcessUserRole")
+          .where
+          .not(valuator_role_id: Decidim::ParticipatoryProcessUserRole.pluck(:id))
+          .destroy_all
+
+        Decidim::Proposals::ValuationAssignment
+          .where(valuator_role_type: "Decidim::AssemblyUserRole")
+          .where
+          .not(valuator_role_id: Decidim::AssemblyUserRole.pluck(:id))
+          .destroy_all
+
+        Decidim::Proposals::ValuationAssignment
+          .where(valuator_role_type: "Decidim::ConferenceUserRole")
+          .where
+          .not(valuator_role_id: Decidim::ConferenceUserRole.pluck(:id))
+          .destroy_all
```

Comment:
On these three cases, we should check if the module is installed before running this command. For instance in a default installation decidim-conferences isn't installed, so you'll see the following error when running this task: 

> NameError: uninitialized constant Decidim::ConferenceUserRole
> 
>           .not(valuator_role_id: Decidim::ConferenceUserRole.pluck(:id))
>                                         ^^^^^^^^^^^^^^^^^^^^
> Did you mean?  Decidim::Conferences

Reply:
I'm picking `decidim-conferences` but as you know there could be installations without `decidim-assemblies` and even without `decidim-participatory_processes`, so we should take that into account